src/code42cli/cmds/search/extraction.py

```diff
@@ -83,13 +80,13 @@ def handle_response(response):
         total_events = len(events)
         handlers.TOTAL_EVENTS += total_events
 
-        def paginate():
-            yield format_func(events, output_header)
+        def _format_output():
+            return formatter.get_formatted_output(events)
 
         if len(events) > 10:
-            click.echo_via_pager(paginate)
+            click.echo_via_pager(_format_output())
         else:
-            for page in paginate():
+            for page in _format_output():
```

Comment:
This is what's causing the extra newlines I mentioned before. It's not just alerts, it's if there are less than 10 events since we're `echo`-ing them individually `echo` is adding a newline like `print` does. 

Changing to `click.echo(page, nl=False)` fixes that.

Reply:
Ah, thanks for finding this!

Reply:
We need to echo a new line at the very end though

Reply:
Oh wait... only if it's table do we need a new line at the end... ugh